documentation/docs/98-reference/21-svelte-reactivity.md

```diff
@@ -22,4 +22,85 @@ Svelte provides reactive versions of various built-ins like `SvelteMap`, `Svelte
 <input bind:value={url.href} />
 ```
 
+The utilities provided in `svelte/reactivity` are automatically reactive with respect to their properties and methods, as seen in the previous example. As such, they don't need to be wrapped in `$state`. However, if a variable is reassigned, it needs to be wrapped in a `$state` in order for this reassignement to be reactive.
+
+```svelte
+<script>
+	import { SvelteURL } from 'svelte/reactivity';
+
+	let url = $state(new SvelteURL('https://example.com/path'));
+</script>
+
+<!-- these are reactive... -->
+Protocol: {url?.protocol ?? "ftp:"}
+<br>
+Hostname: {url?.hostname ?? "svelte.dev"}
+<br>
+Path: {url?.pathname ?? ""}
+
+<hr />
+
+<!-- ...even when reassigning -->
+<button
+	onclick={() => {
+		url = undefined;
+	}}
+>
+	Erase
+</button>
+```
+
+In a similar manner, the values stored inside e.g. `SvelteMap` are not automatically reactive, so if more complex values such as objects are used, they need to be wrapped in a `$state` in order to make their properties reactive as well.
+
+```svelte
+<script>
+	import { SvelteMap } from 'svelte/reactivity';
+
+	const people = new SvelteMap();
+
+	const alice = {name: "Alice", age: 18};
+	people.set("alice", alice);
+
+	const bob = $state({name: "Bob", age: 21});
+	people.set("bob", bob);
+</script>
+
+{#each people.entries() as [id, person] (id)}
+	Name: {person.name}
+	<br>
+	Age: {person.age}
+	<br>
+	<br>
+{/each}
+
+<hr />
+
+<!-- This will NOT propagate reactively -->
+<button
+	onclick={() => {
+		people.get("alice").age++;
+	}}
+>
+	Alice's birthday
+</button>
+
+<!-- This WILL propagate reactively -->
+<button
+	onclick={() => {
+		people.get("bob").age++;
+	}}
+>
+	Bob's birthday
+</button>
+
+<!-- This WILL propagate reactively (but subsequent updates to the object won't) -->
+<button
+	onclick={() => {
+		people.set("carol", {name: "Carol", age: 0});
```

Comment:
would be great to say, maybe needs a separate section on performance but this seems also like a good spot, that for hot paths / frequently updated state, using `$state` for deep reactivity and thus using Proxy's could slow down performance .  I think it's something like 20% slower.  And for such cases, it's probably not a good idea to make every value in a SvelteMap a `$state` signal and just create new objects on updates.

[example](https://svelte.dev/playground/hello-world?version=5.15.0#H4sIAAAAAAAAE51SwYrbMBD9lUEt65g1DtulULy2S3ostBRKT3EOsj1J1HVGxhonDUb_XiQ5JAvppRhZaN7Te6OZmQTJA4pM_OpbyYp28POIHeM32YOuf2PDcJTdiAZOivd6ZHhvWDKCJkDiQaERidiqDo3I1pPgc-_UXEAkF-1V36fGy4pE1NLgvXijiZHYiEzkIZjpnpUmA8NIaGBZVpSbZlA9lxVVrA69Hhimm4wtbAd9gCjcXw4oG1ZHxefoxd3okMFZQwHRSQ9d68NUcaPJMPSo-86BhKer6GLtGBWvI9mpBqMEJqeRQbSaz3KHGTx9spvEMQHWUa3rG94Xf_KsD08zaxP7jNzajtS4d8LoWoA_cDCaFq94TlzpY5iCqkteMR6gmBNNd8iONivxUXZQgP8XBYzU4lYRtvAZJguZA2birBK2e1SHvATX2coEqwSmNE0dnECapkfZ2eBuK8qX195Qvn8u_3-i8uX-2ctM71A2-0sSM7qIQRpYqzaB3tdqAwvVxtbl8d1XfArx1NXfh_N68COz2t2gcvcWDPu0dJbWP6IemTWBpqZTzWsxLWIoyrdtug6F7-_zxwTmrn-VhJGNbRn44In5MmiW_1S_6e2sHbtMHx_h4eGudWxLRc2AByQOJpFxw3ZjdflEIhj_sMh4GNFu7F-AOfRT_QMAAA==)



create a function
```js
	function updatePerson(key, val) {
	  let item = people.get(key);
		val = val === undefined ? {} : val;
		item = item === undefined ? {} : item;
	  people.set(key, {...item, ...val});
	}
```
```svelte
<button onclick={() => people.get('alice').age++ && updatePerson('alice')}>increment alice's age</button>
<button onclick={() => updatePerson('alice', {age: 35, name: 'Jane'})}>update alice</button>
```

Reply:
Yeah, it's probably a good idea to mention that in there and not lead users down the wrong path... Thanks!